core/src/components/item/item.tsx

```diff
@@ -297,7 +297,12 @@ export class Item implements ComponentInterface, AnchorInterface, ButtonInterfac
   private updateCounterOutput(inputEl: HTMLIonInputElement | HTMLIonTextareaElement) {
     if (this.counter && !this.multipleInputs && inputEl?.maxlength !== undefined) {
       const length = inputEl?.value?.toString().length ?? '0';
-      this.counterString = `${length}/${inputEl.maxlength}`;
+      // MD renders the length and total with additional spacing:
+      // https://material.io/components/text-fields/android#using-text-fields
+      // iOS prefers the text contents tight together:
+      // https://uxdesign.cc/ui-cheat-sheet-text-fields-2152112615f8#9233
+      const spacer = getIonMode(this) === 'md' ? ' / ' : '/';
```

Comment:
Is this true? I can't find anything in the Apple Human Interface guidelines that says the character counter `/` should not have spaces.

Reply:
There is no documented spec in Apple Human Interface guidelines. I would imagine because they don't ship a UI component specifically for counters and the pattern is covered in their other guidelines.

This was based on the link in the [comment](https://uxdesign.cc/ui-cheat-sheet-text-fields-2152112615f8#9233). I personally think we should be exposing a function to the implementer to determine how the counter string is formatted. Currently developers are locked to the numeric appearance, which has downsides for UX.

It'd be valuable for an implementer to drive the display as "6 characters remaining", etc.